File: jtak-mobile-master/lib/src/core/models/catalog/restaurant_category_model.dart
class RestaurantCategoryModel {
  final int id;
  final String title;
  final String? titleEn;
  final String image;
  final int order;
  final bool active;
  final String filterTag;

  /// Catalog category this entry stands for. The merchants behind it are the
  /// ones that genuinely sell in it, which is what makes the count meaningful.
  final int? productCategoryId;

  /// How many merchants sell in the category, as reported by the backend.
  final int merchantCount;

  RestaurantCategoryModel({
    required this.id,
    required this.title,
    this.titleEn,
    required this.image,
    required this.order,
    this.active = true,
    required this.filterTag,
    this.productCategoryId,
    this.merchantCount = 0,
  });

  factory RestaurantCategoryModel.fromMap(Map<String, dynamic> map) {
    return RestaurantCategoryModel(
      id: map['id'] is int
          ? map['id']
          : int.tryParse((map['id'] ?? '0').toString()) ?? 0,
      title: (map['title'] ?? '').toString().trim(),
      titleEn: map['titleEn']?.toString().trim(),
      image: (map['image'] ?? '').toString().trim(),
      order: map['order'] is int
          ? map['order']
          : int.tryParse((map['order'] ?? '0').toString()) ?? 0,
      active: map['active'] == true || map['active'] == null,
      filterTag: (map['filterTag'] ?? map['title'] ?? '').toString().trim(),
      productCategoryId: (map['productCategoryId'] as num?)?.toInt(),
      merchantCount: (map['merchantCount'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'title': title,
      'titleEn': titleEn,
      'image': image,
      'order': order,
      'active': active,
      'filterTag': filterTag,
      'productCategoryId': productCategoryId,
      'merchantCount': merchantCount,
    };
  }
}

class RestaurantCategoriesConfigModel {
  final String sectionTitle;
  final String sectionTitleEn;

  /// Heading used where these categories appear as a strip on Home, which
  /// reads differently from the restaurants page listing.
  final String homeSectionTitle;
  final String homeSectionTitleEn;

  final bool enabled;
  final bool showOnHome;
  final List<RestaurantCategoryModel> items;

  RestaurantCategoriesConfigModel({
    this.sectionTitle = 'كل المطاعم',
    this.sectionTitleEn = 'All Restaurants',
    this.homeSectionTitle = 'أصناف متنوعة',
    this.homeSectionTitleEn = 'Browse by kind',
    this.enabled = true,
    this.showOnHome = true,
    required this.items,
  });

  /// Entries worth showing on Home: active, and known to have merchants behind
  /// them. An entry with none can only open an empty list.
  List<RestaurantCategoryModel> get homeItems => items
      .where((item) =>
          item.active &&
          item.title.isNotEmpty &&
          (item.productCategoryId == null || item.merchantCount > 0))
      .toList()
    ..sort((a, b) => a.order.compareTo(b.order));

  factory RestaurantCategoriesConfigModel.fromMap(Map<String, dynamic> map) {
    List rawItems = [];
    if (map['items'] is List) {
      rawItems = map['items'];
    } else if (map['data'] is List) {
      rawItems = map['data'];
    }

    final items = <RestaurantCategoryModel>[];
    for (final item in rawItems) {
      if (item is Map<String, dynamic>) {
        items.add(RestaurantCategoryModel.fromMap(item));
      } else if (item is Map) {
        items.add(
            RestaurantCategoryModel.fromMap(Map<String, dynamic>.from(item)));
      }
    }

    return RestaurantCategoriesConfigModel(
      sectionTitle: (map['sectionTitle'] ?? 'كل المطاعم').toString().trim(),
      sectionTitleEn:
          (map['sectionTitleEn'] ?? 'All Restaurants').toString().trim(),
      homeSectionTitle:
          (map['homeSectionTitle'] ?? 'أصناف متنوعة').toString().trim(),
      homeSectionTitleEn:
          (map['homeSectionTitleEn'] ?? 'Browse by kind').toString().trim(),
      enabled: map['enabled'] != false,
      showOnHome: map['showOnHome'] != false,
      items: items,
    );
  }
}

File: jtak-dashboard-main/src/app/pages/restaurant-categories/models/restaurant-category.model.ts
export interface RestaurantCategoryItem {
  id: number;
  title: string;
  titleEn?: string;
  image: string;
  order: number;
  active: boolean;
  filterTag: string;

  /**
   * Catalog category this entry stands for. This is the real link: the
   * merchants behind it are the ones that actually sell in it. Entries without
   * one fall back to matching on filterTag.
   */
  productCategoryId?: number | null;

  /** Live merchant count from the backend. Zero means customers won't see it. */
  merchantCount?: number;
}

/** A category the admin can point an entry at, with its merchant count. */
export interface RestaurantCategoryTarget {
  id: number;
  title: string;
  parentTitle?: string | null;
  merchantCount: number;
}

export interface RestaurantCategoriesSectionConfig {
  sectionTitle: string;
  sectionTitleEn: string;
  /** Heading for the same categories where they appear as a strip on Home. */
  homeSectionTitle: string;
  homeSectionTitleEn: string;
  enabled: boolean;
  /** Whether the Home strip appears at all. */
  showOnHome: boolean;
  items: RestaurantCategoryItem[];
}

export interface AdminRestaurantCategoriesResponse {
  sectionTitle: string;
  sectionTitleEn: string;
  homeSectionTitle: string;
  homeSectionTitleEn: string;
  enabled: boolean;
  showOnHome: boolean;
  availableCategories: RestaurantCategoryTarget[];
  totalCount: number;
  activeCount: number;
  items: RestaurantCategoryItem[];
}
